pkg/splunk/enterprise/configuration.go

```diff
@@ -1607,3 +1644,71 @@ maxGlobalRawDataSizeMB = %d`, indexDefaults, defaults.MaxGlobalRawDataSizeMB)
 `, indexDefaults)
 	return indexDefaults
 }
+
+// validateProbe validates a generic probe values
+func validateProbe(probe *corev1.Probe) error {
+	if probe.InitialDelaySeconds < 0 || probe.TimeoutSeconds < 0 || probe.PeriodSeconds < 0 || probe.SuccessThreshold < 0 || probe.FailureThreshold < 0 {
+		return fmt.Errorf("Negative values are not not allowed. Configured values InitialDelaySeconds = %d, TimeoutSeconds = %d, PeriodSeconds = %d, SuccessThreshold = %d, FailureThreshold = %d", probe.InitialDelaySeconds, probe.TimeoutSeconds, probe.PeriodSeconds, probe.SuccessThreshold, probe.FailureThreshold)
+	}
+
+	return nil
+}
+
+// validatesLivenessProbe validates the liveness probe config
+func validatesLivenessProbe(ctx context.Context, cr splcommon.MetaObject, livenessProbe *corev1.Probe) error {
+	var err error
+	if livenessProbe == nil {
+		return err
+	}
+
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("validatesLivenessProbe").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	err = validateProbe(livenessProbe)
+	if err != nil {
+		return fmt.Errorf("Invalid Liveness Probe config. Reason: %s", err)
+	}
+
+	if livenessProbe.InitialDelaySeconds < livenessProbeDefaultDelaySec {
```

Comment:
Can't we incorporate all of this logic inside validateProbe by passing in these parameters? Seems like defaultDelaySec, TimeoutSec, PeriodSec are common to ready and liveness probes?

Reply:
The default values we are comparing against are different. I will explore in future on how to refactor and combine these 2 functions.

Reply:
I was thinking default values can be passed to the validateProbe function which both probes can use? Next sprint is fine too :)